### ISSUE TITLE
XMLTV xmltv cache not refreshing, only first one added 

### ISSUE THREAD

**Issue by Oliwerix** [jellyfin/jellyfin#6171]
**Describe the bug**
When adding XMLTV guides, jellyfin only fetches the first one added, and then does not fetch any others, not even on refresh, restart of Jellyfin etc. When mapping the channels only one (the first one added) is available. 
Removing tuner devices or TV guide providers does not fix the issue. Jellyfin still doesn't fetch new ones.

I temporarily fixed it by deleting all contents of /var/cache/jellyfin/xmltv.

**System (please complete the following information):**
 - OS: Ubuntu 20.04 LTS 
 - Virtualization: none
 - Clients: N/A
 - Browser: N/A
 - Jellyfin Version: 10.7.5
 - Playback: N/A
 - Installed Plugins: Playback reporting
 - Reverse Proxy: apache2
 - Base URL: none
 - Networking: host
 - Storage: local

**To Reproduce**
[video](https://of.sijanec.eu/krneki/jelly.mp4)

1. Add m3u source
2. Add xmltv TV guide provider
3. Refresh Guide Data
4. Add another guide provider
5. refresh data
6. Try to map channel 

**Expected behavior**
New channels are expected to show up in mapping options

**Server Logs**

```
2021-06-10 18:03:01.272 +02:00] [INF] Executing "Refresh Guide"
[2021-06-10 18:03:03.682 +02:00] [INF] Found m3u channel: "7201,tvslo1"
[2021-06-10 18:03:03.691 +02:00] [INF] xmltv path: "https://tvprofil.net/xmltv/data/slo1.si/weekly_slo1.si_tvprofil.net.xml"
[2021-06-10 18:03:04.278 +02:00] [INF] Found m3u channel: "7201,tvslo1"
[2021-06-10 18:03:04.278 +02:00] [INF] xmltv path: "https://tvprofil.net/xmltv/data/slo1.si/weekly_slo1.si_tvprofil.net.xml"
[2021-06-10 18:03:04.279 +02:00] [INF] Found epg channel in "XmlTV" null "SLO1" "slo1.si"
[2021-06-10 18:03:04.280 +02:00] [INF] Refreshing guide with 7 days of guide data
[2021-06-10 18:03:04.296 +02:00] [INF] "Refresh Guide" Completed after 0 minute(s) and 3 seconds
[2021-06-10 18:03:04.297 +02:00] [INF] ExecuteQueuedTasks
[2021-06-10 18:03:07.634 +02:00] [INF] Found m3u channel: "7201,tvslo1"
[2021-06-10 18:03:07.637 +02:00] [INF] xmltv path: "https://tvprofil.net/xmltv/data/slo1.si/weekly_slo1.si_tvprofil.net.xml"
[2021-06-10 18:03:07.639 +02:00] [INF] Queuing task "RefreshChannelsScheduledTask"
[2021-06-10 18:03:07.639 +02:00] [INF] Executing "Refresh Guide"
[2021-06-10 18:03:10.643 +02:00] [INF] Found m3u channel: "7201,tvslo1"
[2021-06-10 18:03:10.644 +02:00] [INF] xmltv path: "https://tvprofil.net/xmltv/data/slo1.si/weekly_slo1.si_tvprofil.net.xml"
[2021-06-10 18:03:10.645 +02:00] [INF] Found epg channel in "XmlTV" null "SLO1" "slo1.si"
[2021-06-10 18:03:10.654 +02:00] [INF] Refreshing guide with 7 days of guide data
```
**FFmpeg Logs**

**Browser Console Logs**

**Screenshots**

**Additional context**

The xmltv source and m3u are working after the "fix"

**Comment by Artiume** [comment]
So I've noticed that the EPG is loaded into the new channel if you look at the channel details, but the guide page itself fails to  update.

**Comment by MrChip53** [comment]
Can either of you test that PR I linked by chance? I dont have multiple XMLTV sources to test it but I believe that would fix it.

**Comment by DavorJ** [comment]
@MrChip53, here are some working XMLTV files that you can use to test: https://tvprofil.net/xmltv/ Each XMLTV file contains one channel.

Besides that, it seems Jellyfin is quite picky with XMLTV files. For example, this one contains all the channels and doesn't work for some reason: https://tvprofil.net/xmltv/data/epg_tvprofil.net.xml

**Comment by MrChip53** [comment]
So just to be clear, the issue is that if you have 2 xmltv sources jellyfin will only use the first one?

**Comment by DavorJ** [comment]
> So just to be clear, the issue is that if you have 2 xmltv sources jellyfin will only use the first one?

Indeed. If you use the above 2, you will only see HTV1 during mapping.

> Asking for clarification because I don't understand, after looking at the code, how your "fix" actually did anything lol

I also don't know what Oliwerix meant by the fix.

**Comment by MrChip53** [comment]
> 
> Indeed. If you use the above 2, you will only see HTV1 during mapping.
> 

And what about in the guide? If you use those two sources only htv1 channel will be populated with guide data?

**Comment by MrChip53** [comment]
ok that PR fixes it
